Command 'wipe node_modules' failed with code: 1

When I run the command ```npx react-native clean-project-auto``` I get the following error and the process terminates.

```
Error running 'wipe node_modules': rm: 
Error running 'wipe node_modules': fts_read
Error running 'wipe node_modules': : 
Error running 'wipe node_modules': No such file or directory

❌  Command 'wipe node_modules' failed with code: 1
/Users/sushrut/Documents/GitHub/Desora-Smoker-RN/node_modules/@react-native-community/cli/build/index.js:99
    const message = err.message.replace(/\.$/, '');
                        ^

TypeError: Cannot read property 'message' of undefined
    at handleError (/Users/sushrut/Documents/GitHub/Desora-Smoker-RN/node_modules/@react-native-community/cli/build/index.js:99:25)
    at Command.handleAction (/Users/sushrut/Documents/GitHub/Desora-Smoker-RN/node_modules/@react-native-community/cli/build/index.js:189:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Library Version: 3.6.4
OS: macOS Big Sur 11.5.2

Getting error on react-native clean-project-auto

```
yarn run v1.22.10
$ react-native clean-project-auto

Executing fully-automated project clean.
Use 'react-native clean-project' for more control
Use `./node_modules/.bin/react-native-clean-project' for total control

Error running 'wipe iOS build artifacts': No matching processes belonging to you were found

✅  wipe android build folder task has finished running in 40ms.
✅  wipe temporary caches task has finished running in 32ms.
✅  npm ci task has finished running in 15ms.
✅  watchman cache clear (if watchman is installed) task has finished running in 66ms.
✅  wipe user iOS Pods cache task has finished running in 506ms.
Error running 'clean android project': 

Error running 'clean android project': FAILURE: Build failed with an exception.

* Where:

Error running 'clean android project': Script '/Users/kailashuniyal/Desktop/cwings-delivery-app/node_modules/react-native-unimodules/gradle.groovy' line: 34

* What went wrong:

Error running 'clean android project': A problem occurred evaluating settings 'CwingsDeliveryApp'.
> 
Error running 'clean android project':  (No such file or directory)

* Try:

Error running 'clean android project': Run with --stacktrace
Error running 'clean android project':  option to get the stack trace. Run with 
Error running 'clean android project': --info or --debug option to get more log output. Run with --scan
Error running 'clean android project':  to get full insights.


Error running 'clean android project': * Get more help at https://help.gradle.org


Error running 'clean android project': BUILD FAILED in 6s

❌  Command 'clean android project' failed with code: 1
/Users/kailashuniyal/Desktop/cwings-delivery-app/node_modules/@react-native-community/cli/build/index.js:99
    const message = err.message.replace(/\.$/, '');
                        ^

TypeError: Cannot read property 'message' of undefined
    at handleError (/Users/kailashuniyal/Desktop/cwings-delivery-app/node_modules/@react-native-community/cli/build/index.js:99:25)
    at Command.handleAction (/Users/kailashuniyal/Desktop/cwings-delivery-app/node_modules/@react-native-community/cli/build/index.js:189:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

```
info Fetching system and libraries information...
System:
    OS: macOS 11.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 902.64 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.4.2 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.19.1 - /usr/local/bin/npm
    Watchman: 2021.06.07.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 30.0.2, 30.0.3, 31.0.0
      System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-31 | Google APIs Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.2 AI-202.7660.26.42.7486908
    Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.64.1 => 0.64.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
 ```

my error for 4.0.0-alpha3.0:

(```
node:1870) UnhandledPromiseRejectionWarning: Error: 

Task "clean android project" 
Error: 
FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/adb/dev/REMOVED-PROJECT-NAME/android/settings.gradle' line: 2

* What went wrong:
A problem occurred evaluating settings 'REMOVED-PROJECT-NAME'.
> Could not read script '/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' as it does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
. 
Exit code: 1


    at executeTask (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/internals/executor.js:40:11)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async executeTasks (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/plugin.js:23:11)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1870) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1870) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

my error for 4.0.0-alpha4.0:

```
(node:8151) UnhandledPromiseRejectionWarning: Error: Cannot find module 'chalk'
Require stack:
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/index.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/errors.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/index.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/install/install.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/index.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js
- /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at _chalk (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:9:39)
    at Object.error (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:46:22)
    at handleError (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:101:24)
    at Command.handleAction (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:189:7)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:8151) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:8151) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

now its complaining about some "chalk" library. its not a dependency for the project but i see it in yarn.lock under a bunch of jest dependencies.

> my error for 4.0.0-alpha3.0:
> 
> ```
> node:1870) UnhandledPromiseRejectionWarning: Error: 
> 
> Task "clean android project" 
> Error: 
> FAILURE: Build failed with an exception.
> 
> * Where:
> Settings file '/Users/adb/dev/REMOVED-PROJECT-NAME/android/settings.gradle' line: 2
> 
> * What went wrong:
> A problem occurred evaluating settings 'REMOVED-PROJECT-NAME'.
> > Could not read script '/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' as it does not exist.
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> BUILD FAILED in 4s
> . 
> Exit code: 1
> 
> 
>     at executeTask (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/internals/executor.js:40:11)
>     at processTicksAndRejections (internal/process/task_queues.js:95:5)
>     at async executeTasks (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/plugin.js:23:11)
> (Use `node --trace-warnings ...` to show where the warning was created)
> (node:1870) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> (node:1870) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> ```
> 
> my error for 4.0.0-alpha4.0:
> 
> ```
> (node:8151) UnhandledPromiseRejectionWarning: Error: Cannot find module 'chalk'
> Require stack:
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/index.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/errors.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/index.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/install/install.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/index.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js
> - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native/cli.js
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
>     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
>     at Module.require (internal/modules/cjs/loader.js:974:19)
>     at require (internal/modules/cjs/helpers.js:92:18)
>     at _chalk (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:9:39)
>     at Object.error (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:46:22)
>     at handleError (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:101:24)
>     at Command.handleAction (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:189:7)
> (Use `node --trace-warnings ...` to show where the warning was created)
> (node:8151) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> (node:8151) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> ```
> 
> now its complaining about some "chalk" library. its not a dependency for the project but i see it in yarn.lock under a bunch of jest dependencies.

At which task is the error being thrown?

> > my error for 4.0.0-alpha3.0:
> > ```
> > node:1870) UnhandledPromiseRejectionWarning: Error: 
> > 
> > Task "clean android project" 
> > Error: 
> > FAILURE: Build failed with an exception.
> > 
> > * Where:
> > Settings file '/Users/adb/dev/REMOVED-PROJECT-NAME/android/settings.gradle' line: 2
> > 
> > * What went wrong:
> > A problem occurred evaluating settings 'REMOVED-PROJECT-NAME'.
> > > Could not read script '/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' as it does not exist.
> > 
> > * Try:
> > Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> > 
> > * Get more help at https://help.gradle.org
> > 
> > BUILD FAILED in 4s
> > . 
> > Exit code: 1
> > 
> > 
> >     at executeTask (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/internals/executor.js:40:11)
> >     at processTicksAndRejections (internal/process/task_queues.js:95:5)
> >     at async executeTasks (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/plugin.js:23:11)
> > (Use `node --trace-warnings ...` to show where the warning was created)
> > (node:1870) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> > (node:1870) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > my error for 4.0.0-alpha4.0:
> > ```
> > (node:8151) UnhandledPromiseRejectionWarning: Error: Cannot find module 'chalk'
> > Require stack:
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/index.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/errors.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/index.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/install/install.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/index.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js
> > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native/cli.js
> >     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
> >     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
> >     at Module.require (internal/modules/cjs/loader.js:974:19)
> >     at require (internal/modules/cjs/helpers.js:92:18)
> >     at _chalk (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:9:39)
> >     at Object.error (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:46:22)
> >     at handleError (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:101:24)
> >     at Command.handleAction (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:189:7)
> > (Use `node --trace-warnings ...` to show where the warning was created)
> > (node:8151) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> > (node:8151) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > now its complaining about some "chalk" library. its not a dependency for the project but i see it in yarn.lock under a bunch of jest dependencies.
> 
> At which task is the error being thrown?

clean android, every time

> > > my error for 4.0.0-alpha3.0:
> > > ```
> > > node:1870) UnhandledPromiseRejectionWarning: Error: 
> > > 
> > > Task "clean android project" 
> > > Error: 
> > > FAILURE: Build failed with an exception.
> > > 
> > > * Where:
> > > Settings file '/Users/adb/dev/REMOVED-PROJECT-NAME/android/settings.gradle' line: 2
> > > 
> > > * What went wrong:
> > > A problem occurred evaluating settings 'REMOVED-PROJECT-NAME'.
> > > > Could not read script '/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' as it does not exist.
> > > 
> > > * Try:
> > > Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> > > 
> > > * Get more help at https://help.gradle.org
> > > 
> > > BUILD FAILED in 4s
> > > . 
> > > Exit code: 1
> > > 
> > > 
> > >     at executeTask (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/internals/executor.js:40:11)
> > >     at processTicksAndRejections (internal/process/task_queues.js:95:5)
> > >     at async executeTasks (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native-clean-project/source/plugin.js:23:11)
> > > (Use `node --trace-warnings ...` to show where the warning was created)
> > > (node:1870) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> > > (node:1870) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > my error for 4.0.0-alpha4.0:
> > > ```
> > > (node:8151) UnhandledPromiseRejectionWarning: Error: Cannot find module 'chalk'
> > > Require stack:
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/index.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/errors.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/tools/config/index.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/install/install.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/commands/index.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js
> > > - /Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/react-native/cli.js
> > >     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
> > >     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
> > >     at Module.require (internal/modules/cjs/loader.js:974:19)
> > >     at require (internal/modules/cjs/helpers.js:92:18)
> > >     at _chalk (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:9:39)
> > >     at Object.error (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli-tools/build/logger.js:46:22)
> > >     at handleError (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:101:24)
> > >     at Command.handleAction (/Users/adb/dev/REMOVED-PROJECT-NAME/node_modules/@react-native-community/cli/build/index.js:189:7)
> > > (Use `node --trace-warnings ...` to show where the warning was created)
> > > (node:8151) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
> > > (node:8151) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > now its complaining about some "chalk" library. its not a dependency for the project but i see it in yarn.lock under a bunch of jest dependencies.
> > 
> > 
> > At which task is the error being thrown?
> 
> clean android, every time

Thanks. Now I'm just spitballing, but can you try `rm -r $HOME/.gradle/caches/` and then run `clean-project-auto`?

> A few questions:
> 
> * Which node version are you running?
> * Which `@react-native-community/cli` version do you have?
> * Does this happen every time?
> * At which task does this happen for you? (Some happen on the `wipe node_modules` and some on the `clean android project`)
> 
> Thanks. I can't seem to replicate it but let's get your feedback and fix this asap.

---

@pmadruga I'm just joining into this conversation as another person experiencing the "clean android project" error.

1. Which node version are you running?: **14.13.0, then with 16.13.0**
2. Which @react-native-community/cli version do you have?: **3.6.4**
3. Does this happen every time?: **yes**
4. At which task does this happen for you?: **clean android project**

After the above, I then performed the following using node `16.13.0` and yarn `1.22.17`:

1. upgraded from `3.6.4` to `4.0.0-alpha4.0`
2. wiped and reinstalled node_modules
3. ran `yarn clean` (npm script = `"clean": "react-native clean-project-auto"`)

Here's my CLI output (when the execution broke, node_modules were not found):
 
 ```shell
 Executing fully-automated project clean.
Use 'react-native clean-project' for more control
Use `./node_modules/.bin/react-native-clean-project' for total control


ℹ️  STARTED: "wipe iOS build artifacts"
✅ FINISHED: "wipe iOS build artifacts" task has finished running in 4s.

ℹ️  STARTED: "wipe iOS Pods folder"
✅ FINISHED: "wipe iOS Pods folder" task has finished running in 12ms.

ℹ️  STARTED: "wipe system iOS Pods cache"
✅ FINISHED: "wipe system iOS Pods cache" task has finished running in 4s.

ℹ️  STARTED: "wipe user iOS Pods cache"
✅ FINISHED: "wipe user iOS Pods cache" task has finished running in 11ms.

ℹ️  STARTED: "wipe android build folder"
✅ FINISHED: "wipe android build folder" task has finished running in 10ms.

ℹ️  STARTED: "watchman cache clear (if watchman is installed)"
✅ FINISHED: "watchman cache clear (if watchman is installed)" task has finished running in 66ms.

ℹ️  STARTED: "wipe temporary caches"
✅ FINISHED: "wipe temporary caches" task has finished running in 17ms.

ℹ️  STARTED: "wipe node_modules"
✅ FINISHED: "wipe node_modules" task has finished running in 6s.

ℹ️  STARTED: "yarn cache clean (if yarn is installed)"
✅ FINISHED: "yarn cache clean (if yarn is installed)" task has finished running in 8s.

ℹ️  STARTED: "npm cache verify"
✅ FINISHED: "npm cache verify" task has finished running in 176s.

ℹ️  STARTED: "npm ci"
✅ FINISHED: "npm ci" task has finished running in 13ms.

ℹ️  STARTED: "clean android project"
node:internal/modules/cjs/loader:933
  const err = new Error(message);
              ^

Error: Cannot find module 'chalk'
Require stack:
- /path/to/my/project/node_modules/@react-native-community/cli-tools/build/logger.js
- /path/to/my/project/node_modules/@react-native-community/cli-tools/build/index.js
- /path/to/my/project/node_modules/@react-native-community/cli/build/tools/config/errors.js
- /path/to/my/project/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js
- /path/to/my/project/node_modules/@react-native-community/cli/build/tools/config/index.js
- /path/to/my/project/node_modules/@react-native-community/cli/build/commands/install/install.js
- /path/to/my/project/node_modules/@react-native-community/cli/build/commands/index.js
- /path/to/my/project/node_modules/@react-native-community/cli/build/index.js
- /path/to/my/project/node_modules/react-native/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at _chalk (/path/to/my/project/node_modules/@react-native-community/cli-tools/build/logger.js:9:39)
    at Object.error (/path/to/my/project/node_modules/@react-native-community/cli-tools/build/logger.js:46:22)
    at handleError (/path/to/my/project/node_modules/@react-native-community/cli/build/index.js:101:24)
    at Command.handleAction (/path/to/my/project/node_modules/@react-native-community/cli/build/index.js:189:7) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/path/to/my/project/node_modules/@react-native-community/cli-tools/build/logger.js',
    '/path/to/my/project/node_modules/@react-native-community/cli-tools/build/index.js',
    '/path/to/my/project/node_modules/@react-native-community/cli/build/tools/config/errors.js',
    '/path/to/my/project/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js',
    '/path/to/my/project/node_modules/@react-native-community/cli/build/tools/config/index.js',
    '/path/to/my/project/node_modules/@react-native-community/cli/build/commands/install/install.js',
    '/path/to/my/project/node_modules/@react-native-community/cli/build/commands/index.js',
    '/path/to/my/project/node_modules/@react-native-community/cli/build/index.js',
    '/path/to/my/project/node_modules/react-native/cli.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
 ```

Looks like I'm also in the same boat using automatic cleaning on alpha4.0:

```
yarn run v1.22.5
$ react-native clean-project-auto

Executing fully-automated project clean.
Use 'react-native clean-project' for more control
Use `./node_modules/.bin/react-native-clean-project' for total control


ℹ️  STARTED: "wipe iOS build artifacts"
✅ FINISHED: "wipe iOS build artifacts" task has finished running in 21s.

ℹ️  STARTED: "wipe iOS Pods folder"
✅ FINISHED: "wipe iOS Pods folder" task has finished running in 10ms.

ℹ️  STARTED: "wipe system iOS Pods cache"
✅ FINISHED: "wipe system iOS Pods cache" task has finished running in 5s.

ℹ️  STARTED: "wipe user iOS Pods cache"
✅ FINISHED: "wipe user iOS Pods cache" task has finished running in 17ms.

ℹ️  STARTED: "wipe android build folder"
✅ FINISHED: "wipe android build folder" task has finished running in 12ms.

ℹ️  STARTED: "watchman cache clear (if watchman is installed)"
✅ FINISHED: "watchman cache clear (if watchman is installed)" task has finished running in 25ms.

ℹ️  STARTED: "wipe temporary caches"
✅ FINISHED: "wipe temporary caches" task has finished running in 29ms.

ℹ️  STARTED: "wipe node_modules"
✅ FINISHED: "wipe node_modules" task has finished running in 24s.

ℹ️  STARTED: "yarn cache clean (if yarn is installed)"
✅ FINISHED: "yarn cache clean (if yarn is installed)" task has finished running in 14s.

ℹ️  STARTED: "npm cache verify"
✅ FINISHED: "npm cache verify" task has finished running in 12s.

ℹ️  STARTED: "npm ci"
✅ FINISHED: "npm ci" task has finished running in 9ms.

ℹ️  STARTED: "clean android project"
(node:82573) UnhandledPromiseRejectionWarning: Error: Cannot find module 'chalk'
Require stack:
- /path/to/my-project/node_modules/@react-native-community/cli-tools/build/logger.js
- /path/to/my-project/node_modules/@react-native-community/cli-tools/build/index.js
- /path/to/my-project/node_modules/@react-native-community/cli/build/tools/config/errors.js
- /path/to/my-project/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js
- /path/to/my-project/node_modules/@react-native-community/cli/build/tools/config/index.js
- /path/to/my-project/node_modules/@react-native-community/cli/build/commands/install/install.js
- /path/to/my-project/node_modules/@react-native-community/cli/build/commands/index.js
- /path/to/my-project/node_modules/@react-native-community/cli/build/index.js
- /path/to/my-project/node_modules/react-native/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at _chalk (/path/to/my-project/node_modules/@react-native-community/cli-tools/build/logger.js:9:39)
    at Object.error (/path/to/my-project/node_modules/@react-native-community/cli-tools/build/logger.js:46:22)
    at handleError (/path/to/my-project/node_modules/@react-native-community/cli/build/index.js:101:24)
    at Command.handleAction (/path/to/my-project/node_modules/@react-native-community/cli/build/index.js:189:7)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:82573) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:82573) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

running:
Node: 16
react-native-clean-project: 4.0.0-alpha4.0

running the `react-native clean-project-auto` command.

before installing chalk it fails here:
```
ℹ️  STARTED: "clean android project"
node:internal/modules/cjs/loader:933
  const err = new Error(message);
              ^

Error: Cannot find module 'chalk'
Require stack:
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-tools/build/logger.js
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-tools/build/index.js
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/tools/config/errors.js
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/tools/config/index.js
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/commands/install/install.js
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/commands/index.js
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js
- /Users/adb/dev/APP_NAME/node_modules/react-native/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at _chalk (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-tools/build/logger.js:9:39)
    at Object.error (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-tools/build/logger.js:46:22)
    at handleError (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js:101:24)
    at Command.handleAction (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js:189:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-tools/build/logger.js',
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-tools/build/index.js',
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/tools/config/errors.js',
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/tools/config/readConfigFromDisk.js',
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/tools/config/index.js',
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/commands/install/install.js',
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/commands/index.js',
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js',
    '/Users/adb/dev/APP_NAME/node_modules/react-native/cli.js'
  ]
}
```

so i tried to install chalk(5.0.0). 2 things happen:
1. when i first run the command:
`warn Package chalk has been ignored because it contains invalid configuration. Reason: Package subpath './package.json' is not defined by "exports" in /Users/adb/dev/APP_NAME/node_modules/chalk/package.json`
2. and then again this
```
ℹ️  STARTED: "clean android project"
error 

Task "clean android project" 
Error: 
FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/adb/dev/APP_NAME/android/settings.gradle' line: 2

* What went wrong:
A problem occurred evaluating settings 'APP_NAME'.
> Could not read script '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' as it does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 606ms
. 
Exit code: 1

.
Error: 

Task "clean android project" 
Error: 
FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/adb/dev/APP_NAME/android/settings.gradle' line: 2

* What went wrong:
A problem occurred evaluating settings 'APP_NAME'.
> Could not read script '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' as it does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 606ms
. 
Exit code: 1


    at executeTask (/Users/adb/dev/APP_NAME/node_modules/react-native-clean-project/source/internals/executor.js:40:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async executeTasks (/Users/adb/dev/APP_NAME/node_modules/react-native-clean-project/source/plugin.js:23:11)
    at async Object.func (/Users/adb/dev/APP_NAME/node_modules/react-native-clean-project/source/plugin.js:27:7)
    at async Command.handleAction (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js:186:9)
node:internal/modules/cjs/loader:933
  const err = new Error(message);
              ^

Error: Cannot find module 'chalk'
Require stack:
- /Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js
- /Users/adb/dev/APP_NAME/node_modules/react-native/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at _chalk (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js:16:39)
    at handleError (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js:109:29)
    at Command.handleAction (/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js:189:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/adb/dev/APP_NAME/node_modules/@react-native-community/cli/build/index.js',
    '/Users/adb/dev/APP_NAME/node_modules/react-native/cli.js'
  ]
}
```

however, i did eventually get tired of using the auto command, as i had to reinstall node_modules/ over and over again which took a lot of time, especially if you wanted to use the clean command multiple times per day. so i did switch to using a command which keeps node_modules/.
`./node_modules/.bin/react-native-clean-project --keep-node-modules --remove-iOS-build --remove-android-build --clean-android-project --keep-brew --keep-pods`
when i run this i dont get any errors.

the only thing is that not all commands are able to be specified, so i still have to manually answer some:
```
Wipe iOS Pods folder? (Y/n) n
Wipe system iOS Pods cache? (Y/n) y
Wipe user iOS Pods cache? (Y/n) y
```
i see there is a [discussion to make "keep" and "remove" possible](https://github.com/pmadruga/react-native-clean-project/issues/29), but thats another disucssion.

--- a/source/internals/tasks.js
+++ b/source/internals/tasks.js
@@ -94,11 +94,11 @@ const autoTasks = [
   tasks.wipeAndroidBuildFolder,
   tasks.watchmanCacheClear,
   tasks.wipeTempCaches,
+  tasks.cleanAndroidProject,
   tasks.wipeNodeModules,
   tasks.yarnCacheClean,
   tasks.npmCacheVerify,
   tasks.npmInstall,
-  tasks.cleanAndroidProject
 ];
 
 module.exports = {

